run-unit-test.sh

```diff
@@ -15,5 +15,6 @@ virtualenv env
 . env/bin/activate
 pip install -e .
 pip install nose mock coverage
+rm .coverage
```

Comment:
Hurrah, conclusion reached! This is `nose`'s fault. In [their docs](http://nose.readthedocs.org/en/latest/plugins/cover.html#cmdoption--cover-erase), they document the `--cover-erase` flag. This means they by default use the equivalent of the `-a` flag to `coverage`, which means they deploy a non-default behaviour.

I recommend turning this off in nose with the `--cover-erase` flag.